### PR TITLE
Float 16 without llvm

### PR DESCRIPTION
**Pull request by abadams**
Removed the llvm dependency from float16, which is a barrier to eventually factoring the compiler into IR and codegen components.

Also removed a bunch of unused complexity from the API.

### CODE REVIEW

**Comment by steven-johnson** [comment]
LGTM

**Comment by abadams** [comment]
Looks like support for actually generating code that does half math is unreliable across llvm platforms. Must be new. I'll remove the actual math from the test.

File: src/Float16.h [hunk 1]
```diff
@@ -16,43 +15,19 @@ namespace Halide {
...
  * */
 struct float16_t {
-    // NOTE: Do not use virtual methods here
-    // it will change the size of this data type.
 
     /// \name Constructors
     /// @{
 
-    /** Construct from a float using a particular rounding mode.
+    /** Construct from a float, double, or int using a particular rounding mode.
```
Comment (zvookin) [inline src/Float16.h]: Is "using a particular rounding mode" still true? If the input is expected to be exact anyway, does the rounding mode matter?

File: src/Float16.h [hunk 2]
```diff
@@ -97,93 +72,36 @@ struct float16_t {
...
-
-
     /** Return a new float16_t with a negated sign bit*/
     EXPORT float16_t operator-() const;
 
-    /** \name Overloaded arithmetic operators for convenience
-     * These operators assume RoundingMode::ToNearestTiesToEven rounding
-     */
-    /**@{*/
+    /** Arithmetic operators. */
```
Comment (zvookin) [inline src/Float16.h]: Rounding mode is now unspecified? Do we want to mention this explicitly?

Comment (abadams) [reply src/Float16.h]: Rounding mode is now round to nearest, ties to even. I'll fix the comments.

**Pull request merged**